Vulkan/Android does not compile using NDK 21.3 or NDK 21.1 [in macOS]

```
../../filament/backend/src/vulkan/VulkanHandles.cpp:204:30: error: assigning to 'VkSurfaceKHR' (aka 'unsigned long long') from incompatible type 'nullptr_t'
    surfaceContext.surface = nullptr;
                             ^~~~~~~
1 error generated.
[95/645] Building CXX object filament/backend/CMakeFiles/backend.dir/src/vulkan/VulkanDriver.cpp.o
FAILED: filament/backend/CMakeFiles/backend.dir/src/vulkan/VulkanDriver.cpp.o
/Users/opelykh/Library/Android/sdk/ndk/21.3.6528147/toolchains/llvm/prebuilt/darwin-x86_64/bin/armv7a-linux-androideabi19-clang++ -DFILAMENT_DFG_LUT_SIZE=64 -DFILAMENT_DRIVER_SUPPORTS_VULKAN -DFILAMENT_ENABLE_MATDBG=0 -DFILAMENT_MIN_COMMAND_BUFFERS_SIZE_IN_MB=1 -DFILAMENT_OPENGL_HANDLE_ARENA_SIZE_IN_MB=2 -DFILAMENT_PER_FRAME_COMMANDS_SIZE_IN_MB=1 -DFILAMENT_PER_RENDER_PASS_ARENA_SIZE_IN_MB=2 -DSYSTRACE_TAG=2 -I../../filament/include -Ifilament -I../../filament/src -I../../filament/backend/include -I../../filament/backend/src -I../../libs/math/include -I../../libs/utils/include -I../../third_party/robin-map/tnt/.. -I../../libs/bluevk/include -I../../third_party/vkmemalloc/tnt/../src -mthumb -march=armv7-a -mcpu=cortex-a15 -mfloat-abi=softfp -mfpu=neon-vfpv4 -DANDROID -fPIE -std=c++17 -fstrict-aliasing -Wno-unknown-pragmas -Wno-unused-function -fcolor-diagnostics -DTNT_DEV -g -fstack-protector  -fPIC -Wall -Wextra -Wno-unused-parameter -Wextra-semi -Wnewline-eof -Wdeprecated -Wundef -Wgnu-conditional-omitted-operand -Wweak-vtables -Wnon-virtual-dtor -Wclass-varargs -Wimplicit-fallthrough -Wover-aligned -MD -MT filament/backend/CMakeFiles/backend.dir/src/vulkan/VulkanDriver.cpp.o -MF filament/backend/CMakeFiles/backend.dir/src/vulkan/VulkanDriver.cpp.o.d -o filament/backend/CMakeFiles/backend.dir/src/vulkan/VulkanDriver.cpp.o -c ../../filament/backend/src/vulkan/VulkanDriver.cpp
../../filament/backend/src/vulkan/VulkanDriver.cpp:1431:32: error: cannot initialize a parameter of type 'filament::backend::VulkanDisposer::Key' (aka 'const void *') with an lvalue of type 'VkImage' (aka 'unsigned long long')
    mDisposer.createDisposable(stagingImage, [=] () {
                               ^~~~~~~~~~~~
../../filament/backend/src/vulkan/VulkanDisposer.h:41:31: note: passing argument to parameter 'resource' here
    void createDisposable(Key resource, std::function<void()> destructor) noexcept;
                              ^
../../filament/backend/src/vulkan/VulkanDriver.cpp:1477:31: error: cannot initialize a parameter of type 'filament::backend::VulkanDisposer::Key' (aka 'const void *') with an lvalue of type 'VkImage' (aka 'unsigned long long')
    mDisposer.removeReference(stagingImage);
                              ^~~~~~~~~~~~
```

--- a/filament/backend/src/vulkan/VulkanDriver.cpp
+++ b/filament/backend/src/vulkan/VulkanDriver.cpp
@@ -1428,7 +1428,7 @@ void VulkanDriver::readPixels(Handle<HwRenderTarget> src,
     // Create a disposable to defer execution of the following code until after
     // the work command buffer has completed.
 
-    mDisposer.createDisposable(stagingImage, [=] () {
+    mDisposer.createDisposable((VulkanDisposer::Key) stagingImage, [=] () {
 
         VkImageSubresource subResource { .aspectMask = VK_IMAGE_ASPECT_COLOR_BIT };
         VkSubresourceLayout subResourceLayout;
@@ -1474,7 +1474,7 @@ void VulkanDriver::readPixels(Handle<HwRenderTarget> src,
 
     // Next we reduce the ref count of the image to zero, which schedules the above callback to be
     // executed on the next beginFrame(), after the work command buffer is completed.
-    mDisposer.removeReference(stagingImage);
+    mDisposer.removeReference((VulkanDisposer::Key) stagingImage);
 }
 
 void VulkanDriver::readStreamPixels(Handle<HwStream> sh, uint32_t x, uint32_t y, uint32_t width,

--- a/filament/backend/src/vulkan/VulkanHandles.cpp
+++ b/filament/backend/src/vulkan/VulkanHandles.cpp
@@ -201,7 +201,7 @@ VulkanSwapChain::VulkanSwapChain(VulkanContext& context, VkSurfaceKHR vksurface)
 
 // Headless SwapChain constructor. (does not create a VkSwapChainKHR)
 VulkanSwapChain::VulkanSwapChain(VulkanContext& context, uint32_t width, uint32_t height) {
-    surfaceContext.surface = nullptr;
+    surfaceContext.surface = VK_NULL_HANDLE;
     getHeadlessQueue(context, surfaceContext);
 
     surfaceContext.surfaceFormat.format = VK_FORMAT_R8G8B8A8_UNORM;

